manifests/plugin/default_hostgroup.pp

```diff
@@ -1,4 +1,20 @@
-# Installs foreman_default_hostgroup plugin
-class foreman::plugin::default_hostgroup {
-  foreman::plugin {'default_hostgroup': }
+# @summary This class installs the default_hostgroup plugin and optionally manages the configuration file
+#
+# @param hostgroups An hash of hostgroup names and facts to add to the configuration
+#
+class foreman::plugin::default_hostgroup (
+  Array[Hash] $hostgroups = [],
+){
+  if empty($hostgroups) {
+    $config = undef
+    $config_file = undef
+  } else {
+    $config = template('foreman/default_hostgroup.yaml.erb')
+    $config_file = "${foreman::plugin_config_dir}/default_hostgroup.yaml"
```

Comment:
The config file path uses a prefix `foreman` in the plugin type. Based on the default_hostgroup plugin it expects the file name `default_hostgroup.yaml` and not the `foreman-default_hostgroup.yaml`.